barretenberg/cpp/src/barretenberg/common/ref_array.hpp

```diff
@@ -0,0 +1,139 @@
+#include "barretenberg/common/assert.hpp"
+#include <array>
+#include <cstddef>
+#include <initializer_list>
+#include <iterator>
+#include <stdexcept>
+
+/**
+ * @brief A template class for a reference array. Behaves as if std::array<T&, N> was possible.
+ *
```

Comment:
Not currently used, but originally I was using it and maybe it will come in handy? I thought that having compile time sized arrays was not worth the type complexity vs extra std::vector allocations/dynamicness, but in tight loops it certainly could be